core/query_engine.go

```diff
@@ -112,7 +125,14 @@ func (q *QueryEngine) Sync() error {
 				if _, ok := q.cradle.deps[Where]; !ok {
 					return lucyErr.QueryDependencyNotSatisfied
 				}
-				cradle.Exps.Push(qr.Params.(string))
+
+				param := qr.Params.(string)
+
+				if q.Runtime.CheckForInjection(param) {
```

Comment:
Add an explicit flag for disabling `CheckForInjection` at the top level. Keep it enabled by default though